iceoryx_posh/include/iceoryx_posh/internal/popo/building_blocks/chunk_queue_pusher.inl

```diff
@@ -59,14 +68,12 @@ cxx::expected<ChunkQueueError> ChunkQueuePusher::push(mepoo::SharedChunk chunk)
             auto returnedChunk = mepoo::SharedChunk(chunkManagement);
         }
 
-
-        if (getMembers()->m_semaphoreAttached.load(std::memory_order_acquire) && getMembers()->m_semaphore)
+        if (getMembers()->m_conditionVariableAttached.load(std::memory_order_acquire)
+            && getMembers()->m_conditionVariableDataPtr)
```

Comment:
Maybe we should concentrate on merging #202 as your PR also includes this changes...